src/app/content/PrivateHeader.js

```diff
@@ -2,45 +2,84 @@ import React from 'react';
 
-const PrivateHeader = ({ search, onSearch, location }) => {
+import ContactModal from '../components/ContactModal';
+
+const PrivateHeader = ({ title, search, onSearch, location, expanded, onToggleExpand }) => {
     const [{ hasPaidMail }] = useUser();
+    const { createModal } = useModals();
     const inSettings = location.pathname.startsWith('/contacts/settings');
+    const activeBreakpoint = useActiveBreakpoint();
+    const isMobile = isItMobile(activeBreakpoint);
+
     return (
         <header className="header flex flex-nowrap reset4print">
-            <MainLogo url="/contacts" />
-            {!inSettings && <Searchbox placeholder={c('Placeholder').t`Search`} value={search} onChange={onSearch} />}
+            <MainLogo url="/contacts" className="nomobile" />
+            <Hamburger expanded={expanded} onToggle={onToggleExpand} />
+            {title && isMobile ? <span className="big ellipsis">{title}</span> : null}
+            {inSettings || isMobile ? null : (
+                <Searchbox placeholder={c('Placeholder').t`Search`} value={search} onChange={onSearch} />
+            )}
             <TopNavbar>
-                {hasPaidMail ? null : <UpgradeButton external={true} />}
-                <TopNavbarLink
-                    to="/contacts"
-                    icon="contacts"
-                    text={c('Title').t`Contacts`}
-                    aria-current={!inSettings}
-                />
-                <TopNavbarLink
-                    to="/contacts/settings"
-                    icon="settings-master"
-                    text={c('Title').t`Settings`}
-                    aria-current={inSettings}
-                />
-                <SupportDropdown />
+                {hasPaidMail || isMobile ? null : <UpgradeButton external={true} />}
+                {isMobile && !inSettings ? null : (
+                    <TopNavbarLink
+                        className="nomobile"
+                        to="/contacts"
+                        icon="contacts"
+                        text={c('Title').t`Contacts`}
+                        aria-current={!inSettings}
+                    />
+                )}
+                {!inSettings && isMobile ? (
+                    <SearchDropdown
+                        content={
+                            <Icon
+                                name="search"
+                                size={25}
+                                className="topnav-icon mr0-5 flex-item-centered-vert fill-white"
+                            />
+                        }
+                        placeholder={c('Placeholder').t`Search contacts`}
+                        search={search}
+                        onSearch={onSearch}
+                        hasCaret={false}
+                    />
+                ) : null}
+                {isMobile && inSettings ? null : (
+                    <TopNavbarLink
+                        to="/contacts/settings"
```

Comment:
This icon needs to be size 25 as well. Currently it's not possible to do that with the `TopNavbarLink` component. We need to add maybe a `size` prop there, or allow the `icon` prop to be either a node or a string

Reply:
This is managed by the `design-system` right @nico3333fr ?